lib/mix/lib/mix/tasks/compile.app.ex

```diff
@@ -82,6 +82,8 @@ defmodule Mix.Tasks.Compile.App do
   ## Command line options
 
     * `--force` - forces compilation regardless of modification times
+    * `--path path` - where to find `.beam` files and write the resulting
+      `.app` file, defaults to `Mix.Project.compile_path/0`
```

Comment:
Because many things call "compile" passing all args, I am afraid of accidentally passing a --path. Maybe we can call it `--compile-path`?